File: 506-relative-ranks/506-relative-ranks.js
/**
 * @param {number[]} score
 * @return {string[]}
 */
var findRelativeRanks = function(score) {
    let answer = [];
    let saved = [...score];
    score.sort((a,b) => b - a);
    let hash = {};
    hash[score[0]] = "Gold Medal";
    hash[score[1]] = "Silver Medal";
    hash[score[2]] = "Bronze Medal";
    for (let i = 3; i < score.length; i++){
        hash[score[i]] = `${i + 1}`;
    }
    for (let i = 0; i < saved.length; i++){
        answer.push(hash[saved[i]]);
    }
    return answer;
};

File: 506-relative-ranks/506-relative-ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        sorted_score = sorted(list(range(len(score))), key = lambda n: -score[n])
        for i, num in enumerate(sorted_score):
            if i == 0: score[num] = "Gold Medal"
            elif i == 1: score[num] = "Silver Medal"
            elif i == 2: score[num] = "Bronze Medal"
            else: score[num] = str(i+1)
        return score